<b>Name</b>: Bahul Jain<br>
<b>UNI</b>: bkj2111

In [2]:
ls

E6040_HW0.ipynb


In [3]:
cd ..

/home/ubuntu


In [4]:
ls

Deep-Learning/           DeepLearningTutorials/  mycert.pem  README.md  yes.pub
Deep-Learning-Columbia/  miniconda2/             mykey.key   yes


In [5]:
cd DeepLearningTutorials/

/home/ubuntu/DeepLearningTutorials


In [6]:
cd code/

/home/ubuntu/DeepLearningTutorials/code


In [7]:
ls

cA.py                 hmc/                logistic_sgd.py  rnnrbm.py  utils.py
convolutional_mlp.py  imdb_preprocess.py  lstm.py          rnnslu.py
dA.py                 imdb.py             mlp.py           SdA.py
DBN.py                logistic_cg.py      rbm.py           test.py


<b>Problem 1</b>: Clearly GPU for the WIN!!!

In [15]:
%run logistic_sgd.py

... loading data
... building the model
... training the model
epoch 1, minibatch 83/83, validation error 12.458333 %
     epoch 1, minibatch 83/83, test error of best model 12.375000 %
epoch 2, minibatch 83/83, validation error 11.010417 %
     epoch 2, minibatch 83/83, test error of best model 10.958333 %
epoch 3, minibatch 83/83, validation error 10.312500 %
     epoch 3, minibatch 83/83, test error of best model 10.312500 %
epoch 4, minibatch 83/83, validation error 9.875000 %
     epoch 4, minibatch 83/83, test error of best model 9.833333 %
epoch 5, minibatch 83/83, validation error 9.562500 %
     epoch 5, minibatch 83/83, test error of best model 9.479167 %
epoch 6, minibatch 83/83, validation error 9.322917 %
     epoch 6, minibatch 83/83, test error of best model 9.291667 %
epoch 7, minibatch 83/83, validation error 9.187500 %
     epoch 7, minibatch 83/83, test error of best model 9.000000 %
epoch 8, minibatch 83/83, validation error 8.989583 %
     epoch 8, minibatch 83/83,

KeyboardInterrupt: 

<h3>Stochastic Logistic Regression using GPU</h3>

<b>Time</b>: 6.4s<br>
<b>Epoch</b>: 73<br>
<b>Validation Error</b>: 7.49 %<br>

In [16]:
%run convolutional_mlp.py

... loading data
... building the model
... training
training @ iter =  0
epoch 1, minibatch 100/100, validation error 9.230000 %
     epoch 1, minibatch 100/100, test error of best model 9.540000 %
training @ iter =  100
epoch 2, minibatch 100/100, validation error 6.170000 %
     epoch 2, minibatch 100/100, test error of best model 6.480000 %
training @ iter =  200
epoch 3, minibatch 100/100, validation error 4.640000 %
     epoch 3, minibatch 100/100, test error of best model 4.840000 %
training @ iter =  300
epoch 4, minibatch 100/100, validation error 3.500000 %
     epoch 4, minibatch 100/100, test error of best model 3.930000 %
training @ iter =  400
epoch 5, minibatch 100/100, validation error 3.030000 %
     epoch 5, minibatch 100/100, test error of best model 3.270000 %
training @ iter =  500
epoch 6, minibatch 100/100, validation error 2.810000 %
     epoch 6, minibatch 100/100, test error of best model 2.810000 %
training @ iter =  600
epoch 7, minibatch 100/100, validation

KeyboardInterrupt: 

<h3>Convolutional MLP using GPU</h3>

<b>Time</b>: 60s<br>
<b>Epoch</b>: 23<br>
<b>Iterations</b>: 2300<br>
<b>Validation Error</b>: 1.43 %<br>

In [4]:
%env THEANO_FLAGS=device=cpu

env: THEANO_FLAGS=device=cpu


In [5]:
ls

E6040_HW0.ipynb


In [6]:
cd ..


/home/ubuntu


In [7]:
cd DeepLearningTutorials/code/

/home/ubuntu/DeepLearningTutorials/code


In [8]:
%run logistic_sgd.py

... loading data
... building the model
... training the model
epoch 1, minibatch 83/83, validation error 12.458333 %
     epoch 1, minibatch 83/83, test error of best model 12.375000 %
epoch 2, minibatch 83/83, validation error 11.010417 %
     epoch 2, minibatch 83/83, test error of best model 10.958333 %
epoch 3, minibatch 83/83, validation error 10.312500 %
     epoch 3, minibatch 83/83, test error of best model 10.312500 %
epoch 4, minibatch 83/83, validation error 9.875000 %
     epoch 4, minibatch 83/83, test error of best model 9.833333 %
epoch 5, minibatch 83/83, validation error 9.562500 %
     epoch 5, minibatch 83/83, test error of best model 9.479167 %
epoch 6, minibatch 83/83, validation error 9.322917 %
     epoch 6, minibatch 83/83, test error of best model 9.291667 %
epoch 7, minibatch 83/83, validation error 9.187500 %
     epoch 7, minibatch 83/83, test error of best model 9.000000 %
epoch 8, minibatch 83/83, validation error 8.989583 %
     epoch 8, minibatch 83/83,

The code for file logistic_sgd.py ran for 21.5s


<h3>Stochastic Logistic Regression using CPU</h3>

<b>Time</b>: 21.5s<br>
<b>Epoch</b>: 73<br>
<b>Validation Error</b>: 7.49 %<br>

In [14]:
%run convolutional_mlp.py

... loading data
... building the model
... training
training @ iter =  0
epoch 1, minibatch 100/100, validation error 9.230000 %
     epoch 1, minibatch 100/100, test error of best model 9.520000 %
training @ iter =  100


KeyboardInterrupt: 

<h3>Convolutional MLP using CPU</h3>

<b>Time</b>: over 2 mins<br>
<b>Epoch</b>: 1<br>
<b>Iterations</b>: ~200<br>
<b>Validation Error</b>: 9.5 %<br>